src/lower/lowerer_impl_imperative.cpp

```diff
@@ -3568,12 +3594,32 @@ Stmt LowererImplImperative::codeToIncIteratorVars(Expr coordinate, IndexVar coor
   for (auto& iterator : levelIterators) {
     Expr ivar = iterator.getIteratorVar();
     if (iterator.isUnique()) {
-      Expr increment = iterator.isFull()
-                     ? 1
-                     : ir::Cast::make(Eq::make(iterator.getCoordVar(),
-                                               coordinate),
-                                      ivar.type());
-      result.push_back(compoundAssign(ivar, increment));
+      // BEGIN GJ
+      if (iterator.isFull()) {
+        Expr increment = 1;
+        result.push_back(compoundAssign(ivar, increment));
+      } else {
+        Expr iteratorParentPos = iterator.getParent().getPosVar();
+        ModeFunction iterBounds = iterator.posBounds(iteratorParentPos);
+        result.push_back(iterBounds.compute());
+        auto iteratorVar = iterator.getIteratorVar();
+        vector<Expr> iterBSArgs = {
```

Comment:
[nit]: variable name `iterBSArgs` should probably be something like `gallopArgs`.